fontc/src/lib.rs

```diff
@@ -201,28 +208,14 @@ pub fn require_dir(dir: &Path) -> Result<(), Error> {
 }
 
 pub fn init_paths(
-    output_file: Option<&PathBuf>,
     build_dir: &Path,
     flags: Flags,
-) -> Result<(IrPaths, BePaths), Error> {
+) -> Result<(Option<IrPaths>, Option<BePaths>), Error> {
     let ir_paths = IrPaths::new(build_dir);
-    let be_paths = if let Some(output_file) = output_file {
-        BePaths::with_output_file(build_dir, output_file)
-    } else {
-        BePaths::new(build_dir)
-    };
-    // create the output file's parent directory if it doesn't exist
-    if let Some(output_file) = output_file
-        && let Some(parent) = output_file.parent()
-    {
-        require_dir(parent)?;
-    }
+    let be_paths = BePaths::new(build_dir);
 
-    // the build dir stores the IR (for incremental builds) and the default output
-    // file ('font.ttf') so we don't need to create one unless we're writing to it
-    if output_file.is_none() || flags.contains(Flags::EMIT_IR) {
-        require_dir(build_dir)?;
```

Comment:
this is actually a regression. When --emit-ir is set but the build directory doesn't exist, fontc fails because it tries to create subdirectories without first creating the parent.

```
$ git checkout paths-all-optional
$ cargo build --release -p fontc
$ rm -rf build
$ target/release/fontc --emit-ir resources/testdata/static.designspace
[2025-12-02T16:49:23.748060Z ThreadId(1) fontc ERROR] io failed for 'build/anchor_ir': 'No such file or directory (os error 2)'
```